modules/ingester/instance_test.go

```diff
@@ -255,6 +256,90 @@ func TestInstanceLimits(t *testing.T) {
 	}
 }
 
+func TestInstanceCutCompleteTraces(t *testing.T) {
+	tempDir, _ := ioutil.TempDir("/tmp", "")
+	defer os.RemoveAll(tempDir)
+
+	id := make([]byte, 16)
+	rand.Read(id)
+	tracepb := test.MakeTrace(10, id)
+	pastTrace := &trace{
+		traceID:    id,
+		trace:      tracepb,
+		lastAppend: time.Now().Add(-time.Hour),
+	}
+
+	id = make([]byte, 16)
+	rand.Read(id)
+	nowTrace := &trace{
+		traceID:    id,
+		trace:      tracepb,
+		lastAppend: time.Now().Add(time.Hour),
+	}
+
+	tt := []struct {
+		name             string
+		cutoff           time.Duration
+		immediate        bool
+		input            []*trace
+		expectedExist    []*trace
+		expectedNotExist []*trace
+	}{
+		{
+			name:      "empty",
+			cutoff:    0,
+			immediate: false,
+		},
+		{
+			name:             "cut immediate",
+			cutoff:           0,
+			immediate:        true,
+			input:            []*trace{pastTrace, nowTrace},
+			expectedNotExist: []*trace{pastTrace, nowTrace},
+		},
+		{
+			name:             "cut recent",
+			cutoff:           0,
+			immediate:        false,
+			input:            []*trace{pastTrace, nowTrace},
+			expectedExist:    []*trace{nowTrace},
+			expectedNotExist: []*trace{pastTrace},
+		},
+		{
+			name:             "cut all time",
+			cutoff:           2 * time.Hour,
+			immediate:        false,
+			input:            []*trace{pastTrace, nowTrace},
+			expectedNotExist: []*trace{pastTrace, nowTrace},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			instance := defaultInstance(t, tempDir)
+
+			for _, trace := range tc.input {
+				fp := instance.tokenForTraceID(trace.traceID)
+				instance.traces[fp] = trace
+			}
+
+			err := instance.CutCompleteTraces(tc.cutoff, tc.immediate)
+			assert.NoError(t, err)
```

Comment:
I think you should use testify/require here, if err is not nil, the test should stop at this point

Reply:
agreed.